apollo-ios-pagination/Sources/ApolloPagination/GraphQLQueryPager.swift

```diff
@@ -2,204 +2,342 @@ import Apollo
 import ApolloAPI
 import Combine
 import Foundation
+import OrderedCollections
+
+public protocol PagerType {
+  associatedtype InitialQuery: GraphQLQuery
+  associatedtype PaginatedQuery: GraphQLQuery
+  typealias Output = (InitialQuery.Data, [PaginatedQuery.Data], UpdateSource)
+
+  var canLoadNext: Bool { get }
+  func cancel()
+  func loadMore(
+    cachePolicy: CachePolicy,
+    completion: (@MainActor () -> Void)?
+  ) throws
+  func refetch(cachePolicy: CachePolicy)
+  func fetch()
+}
 
 /// Handles pagination in the queue by managing multiple query watchers.
-public class GraphQLQueryPager<InitialQuery: GraphQLQuery, PaginatedQuery: GraphQLQuery> {
+public class GraphQLQueryPager<InitialQuery: GraphQLQuery, PaginatedQuery: GraphQLQuery>: PagerType {
+  public typealias Output = (InitialQuery.Data, [PaginatedQuery.Data], UpdateSource)
+
+  private let pager: Actor
+  private var cancellables: [AnyCancellable] = []
+  private var canLoadNextSubject: CurrentValueSubject<Bool, Never> = .init(false)
 
   /// The result of either the initial query or the paginated query, for the purpose of extracting a `PageInfo` from it.
   public enum PageExtractionData {
+    // This class is outside of the scope of the `GraphQLQueryPager` such that it can be shared between it and the `Actor`.
```

Comment:
Maybe I'm missing something... What is the "this class" that this line of documentation is referring to?

Reply:
Ah, removing that doc -- it's not valid anymore.